app/models/katello/concerns/pulp_database_unit.rb

```diff
@@ -194,7 +194,12 @@ def db_columns_copy
 
       def db_values_copy(source_repo, dest_repo)
         db_values = []
-        new_units = self.repository_association_class.where(repository: source_repo).where.not(unit_id_field => self.repository_association_class.where(repository: dest_repo).pluck(unit_id_field))
+        existing_unit_ids = self.repository_association_class.where(repository: dest_repo).pluck(unit_id_field)
+        if existing_unit_ids.empty?
+          new_units = self.repository_association_class.where(repository: source_repo)
+        else
+          new_units = self.repository_association_class.where(repository: source_repo).where.not("#{unit_id_field} in (?) ", existing_unit_ids)
```

Comment:
Had to add this block because the whole query returns []  for `where.not("#{unit_id_field} in (?) ", existing_unit_ids)` if existing_unit_ids is empty.